pandas/core/indexes/multi.py

```diff
@@ -840,6 +840,46 @@ def size(self) -> int:
 
     @cache_readonly
     def levels(self) -> FrozenList:
+        """
+        Levels of the MultiIndex.
+
+        What are levels?
+        Levels refer to the different hierarchical levels or layers in a MultiIndex.
+        In a MultiIndex, each level represents a distinct dimension or category of
+        the index.
+
+        How can levels be seen in pandas?
+        To access the levels, you can use the levels attribute of the MultiIndex,
+        which returns a tuple of Index objects. Each Index object represents a
+        level in the MultiIndex and contains the unique values found in that
+        specific level.
+
+        Notes
+        -----
+        If the MultiIndex is sliced, this method still returns the original
+        levels of the MultiIndex.
+        When using this method on a DataFrame index, it may show "extra".
+
+        Examples
+        --------
+        Example 1:
+        >>> idx = pd.MultiIndex.from_product([(0, 1, 2), ('green', 'purple')],
+        ...                                  names=['number', 'color'])
+        >>> idx.levels
+        FrozenList([[0, 1, 2], ['green', 'purple']])
+
+        Example 2:
+        >>> idx = pd.MultiIndex.from_product([['John', 'Josh', 'Alex'],
+        ... list('abcde')], names=['Person', 'Letter'])
+        >>> large = pd.DataFrame(data=np.random.randn(15, 2),
+        ... index=idx, columns=['one', 'two'])
+        >>> small = large.loc[['Jo'==d[0:2] for d in
```

Comment:
Can you show the dataframe here first, so the user can better understand the data of the example? I think you'll want to avoid using random data in the example, as it'll make your life more difficult. And if you use data that is meaningful instead of some random things, that will also help users understand better. As an idea, you can use a MultiIndex of animals with class/animal (e.g. insect/ant, insect/spides, mammal/goat...) and use a dataframe with just one column, for example number of legs (we use this example in other docs). The example is silly, but meaningful and any user can understand it very quickly, and focus on the concepts, not in understanding the data.

Then, as said, you can show the dataframe first, show the levels of the multiindex from it (i.e. `df.index.levels`), and then you filter the data `many_leg_animals = animals[animals.num_legs > 4]` to finally show that even if now there are no mammals, in the data, the multiindex still contains all the levels.

Reply:
<img width="557" alt="屏幕截图 2023-10-08 164751" src="https://github.com/pandas-dev/pandas/assets/143710553/6ae56754-4c39-4620-b800-d38de88f2ebe">

Thank you for your suggestion. Is it easier to understand the example part?

Reply:
I have a problem that I may need your help with. When I tested the code after I committed it, four tests failed, but I couldn't figure out exactly what went wrong from the error log provided.